README.md
# card_horizontal_swipe

A new Flutter package project.

## Getting Started

This project is a starting point for a Dart
[package](https://flutter.dev/developing-packages/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://flutter.dev/docs), which offers tutorials, 
samples, guidance on mobile development, and a full API reference.


CHANGELOG.md
## [0.0.1] - 23-04-2021.

* flutter package for animated swipe cards first release.


test/card_horizontal_swipe_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:card_horizontal_swipe/card_horizontal_swipe.dart';

void main() {
  test('adds one to input values', () {
    final calculator = Calculator();
    expect(calculator.addOne(2), 3);
    expect(calculator.addOne(-7), -6);
    expect(calculator.addOne(0), 1);
    expect(() => calculator.addOne(null), throwsNoSuchMethodError);
  });
}


horizontal_scroll_example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:horizontal_scroll_example/main.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(MyApp());

    // Verify that our counter starts at 0.
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Tap the '+' icon and trigger a frame.
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();

    // Verify that our counter has incremented.
    expect(find.text('0'), findsNothing);
    expect(find.text('1'), findsOneWidget);
  });
}


lib/card_horizontal_swipe.dart
library card_horizontal_swipe;

import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';


class CircularListView extends StatelessWidget {
  final Axis scrollDirection;
  final List<Widget> children;
  final ScrollController controller;
  final ScrollPhysics physics;
  final double diameterRatio;
  final double perspective;
  final double offAxisFraction;
  final bool useMagnifier;
  final double magnification;
  final double overAndUnderCenterOpacity;
  final double itemExtent;
  final double squeeze;
  final ValueChanged<int> onSelectedItemChanged;
  final bool renderChildrenOutsideViewport;
  final ListWheelChildDelegate childDelegate;
  final Clip clipBehavior;

  const CircularListView({
    Key key,
    this.scrollDirection = Axis.horizontal,
    this.controller,
    this.physics,
    this.diameterRatio = RenderListWheelViewport.defaultDiameterRatio,
    this.perspective = RenderListWheelViewport.defaultPerspective,
    this.offAxisFraction = 0.0,
    this.useMagnifier = false,
    this.magnification = 1.0,
    this.overAndUnderCenterOpacity = 1.0,
    @required this.itemExtent,
    this.squeeze = 1.0,
    this.onSelectedItemChanged,
    this.renderChildrenOutsideViewport = false,
    this.clipBehavior = Clip.hardEdge,
    @required this.children,
  })  : childDelegate = null,
        super(key: key);

 const CircularListView.useDelegate({
    Key key,
    this.scrollDirection = Axis.vertical,
    this.controller,
    this.physics,
    this.diameterRatio = RenderListWheelViewport.defaultDiameterRatio,
    this.perspective = RenderListWheelViewport.defaultPerspective,
    this.offAxisFraction = 0.0,
    this.useMagnifier = false,
    this.magnification = 1.0,
    this.overAndUnderCenterOpacity = 1.0,
    @required this.itemExtent,
    this.squeeze = 1.0,
    this.onSelectedItemChanged,
    this.renderChildrenOutsideViewport = false,
    this.clipBehavior = Clip.hardEdge,
    @required this.childDelegate,
  })  : children = null,
        super(key: key);

  @override 
  Widget build(BuildContext context) {
    final _childDelegate = children != null
        ? ListWheelChildListDelegate(
            children: children.map((child) {
            return RotatedBox(
              quarterTurns: scrollDirection == Axis.horizontal ? 1 : 0,
              child: child,
            );
          }).toList())
        : ListWheelChildBuilderDelegate(
            builder: (context, index) {
              return RotatedBox(
                quarterTurns: scrollDirection == Axis.horizontal ? 1 : 0,
                child: childDelegate.build(context, index),
              );
            },
          );

    return RotatedBox(
      quarterTurns: scrollDirection == Axis.horizontal ? 3 : 0,
      child: ListWheelScrollView.useDelegate(
        controller: controller,
        physics: FixedExtentScrollPhysics(),
        diameterRatio: diameterRatio,
        perspective: perspective,
        offAxisFraction: offAxisFraction,
        useMagnifier: useMagnifier,
        magnification: magnification,
        overAndUnderCenterOpacity: overAndUnderCenterOpacity,
        itemExtent: itemExtent,
        squeeze: squeeze,
        
        onSelectedItemChanged: onSelectedItemChanged,
        renderChildrenOutsideViewport: renderChildrenOutsideViewport,
        clipBehavior: clipBehavior,
        childDelegate: _childDelegate,
      ),
    );
  }
}


horizontal_scroll_example/lib/horizontal_scroll_with_builder.dart
import 'package:card_horizontal_swipe/card_horizontal_swipe.dart';
import 'package:flutter/material.dart';

class HorizontalWithBuilder extends StatelessWidget {
  const HorizontalWithBuilder({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text('Default'),
        ),
        body: Center(
          child: Container(
            height: 100,
            color: Colors.grey[400],
            child: CircularListView.useDelegate(
              scrollDirection: Axis.horizontal,
              itemExtent: 80,
              magnification: 1.5,
              useMagnifier: true, 
              onSelectedItemChanged: _onItemSelectedChanged,
              childDelegate: ListWheelChildBuilderDelegate(
                  builder: (BuildContext context, int index) {
                return Container(
                  margin: EdgeInsets.symmetric(horizontal: 5),
                  color:index%2==0? Colors.red:Colors.yellow,
                  child: Center(child: Text("Card $index")),
                  );
              }),
            ),
          ),
        ),
      ),
    );
  }

  void _onItemSelectedChanged(int value) {
    print("index : $value");
  }
}

horizontal_scroll_example/lib/horizontal_scroll_list.dart
import 'package:card_horizontal_swipe/card_horizontal_swipe.dart';
import 'package:flutter/material.dart';
import 'package:horizontal_scroll_example/colored_container.dart';

class HorizontalScrollList extends StatelessWidget {
  const HorizontalScrollList({Key key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
          home: Scaffold(
        appBar: AppBar(
          title: Text('Default'),
        ),
        body: Center(
          child: Container(
            height: 150,
            color: Colors.grey[400],
            child: CircularListView(
              scrollDirection: Axis.horizontal,
              itemExtent: 80,
              magnification: 1.5,
              useMagnifier: true,
              children: [
                ColoredContainer(Colors.red),
                ColoredContainer(Colors.yellow),
                ColoredContainer(Colors.green),
                ColoredContainer(Colors.red),
                ColoredContainer(Colors.yellow),
                ColoredContainer(Colors.green),
                ColoredContainer(Colors.red),
                ColoredContainer(Colors.yellow),
                ColoredContainer(Colors.green),
              ],
            ),
          ),
        ),
      ),
    );
  }
}